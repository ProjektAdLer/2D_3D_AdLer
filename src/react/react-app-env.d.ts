/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
