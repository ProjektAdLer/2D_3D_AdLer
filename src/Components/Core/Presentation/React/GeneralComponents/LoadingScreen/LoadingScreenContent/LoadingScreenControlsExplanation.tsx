import { useTranslation } from "react-i18next";
import ControlsExplanationContent from "~ReactComponents/GeneralComponents/ControlsExplanationModal/ControlsExplanationContent";

export default function LoadingScreenControlsExplanation() {
  const { t: translate } = useTranslation("learningSpace");

  return (
    <div className="row-span-4 flex max-h-[45vh] w-[90vw] max-w-[90vw] flex-col overflow-y-auto overflow-x-hidden rounded-xl bg-buttonbgblue p-4 lg:max-h-[60vh] xl:max-h-[50vh] mobile-portrait:h-[60vh] mobile-portrait:max-h-[65vh] portrait:h-[50vh]">
      <h1 className="flex-shrink-0 font-bold text-adlerdarkblue portrait:pb-6">
        {translate("sidebar_controls")}
      </h1>

      <div className="flex min-h-0 flex-1 flex-col">
        <div className="flex-1 overflow-y-auto pb-4">
          <ControlsExplanationContent />
        </div>

        <div className="flex-shrink-0 border-t border-adlerdarkblue/20 pt-4">
          <p className="w-full text-center text-xs font-bold text-adlerdarkblue mobile-landscape:text-2xs portrait:text-2xs">
            {translate("hint_controlsExplanationModal")}
          </p>
        </div>
      </div>
    </div>
  );
}
