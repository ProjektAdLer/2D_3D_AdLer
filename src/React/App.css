body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
