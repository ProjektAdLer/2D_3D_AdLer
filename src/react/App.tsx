import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
