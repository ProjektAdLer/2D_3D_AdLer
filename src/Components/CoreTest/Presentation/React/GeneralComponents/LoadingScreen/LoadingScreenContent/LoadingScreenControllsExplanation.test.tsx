import React from "react";
import { render } from "@testing-library/react";
import LoadingScreenControlsExplanation from "../../../../../../Core/Presentation/React/GeneralComponents/LoadingScreen/LoadingScreenContent/LoadingScreenControlsExplanation";

describe("LoadingScreenControlsExplanation", () => {
  test("should render", () => {
    const { container } = render(<LoadingScreenControlsExplanation />);
    expect(container).toMatchSnapshot();
  });
});
