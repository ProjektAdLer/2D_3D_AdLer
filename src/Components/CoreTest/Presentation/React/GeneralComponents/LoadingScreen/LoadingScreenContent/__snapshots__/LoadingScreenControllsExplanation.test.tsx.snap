// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreenControlsExplanation should render 1`] = `
<div>
  <div
    class="row-span-4 flex max-h-[45vh] w-[90vw] max-w-[90vw] flex-col overflow-y-auto overflow-x-hidden rounded-xl bg-buttonbgblue p-4 lg:max-h-[60vh] xl:max-h-[50vh] mobile-portrait:h-[60vh] mobile-portrait:max-h-[65vh] portrait:h-[50vh]"
  >
    <h1
      class="flex-shrink-0 font-bold text-adlerdarkblue portrait:pb-6"
    >
      sidebar_controls
    </h1>
    <div
      class="flex min-h-0 flex-1 flex-col"
    >
      <div
        class="flex-1 overflow-y-auto pb-4"
      >
        <div
          class="max-w-[85vw] pb-4 text-xs lg:text-sm mobile-landscape:text-2xs portrait:text-xs"
        >
          <div
            class="space-y-6 mobile-landscape:space-y-4 portrait:space-y-4"
          >
            <section
              class="space-y-3 mobile-landscape:space-y-2 portrait:space-y-2"
            >
              <h3
                class="font-bold text-adlerdarkblue"
              >
                controlAvatar
              </h3>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Linksklick"
                  class="w-6 flex-shrink-0"
                  src="left-click.svg"
                />
                <p>
                  controlAvatarMouse
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Einfacher Touch"
                  class="w-6 flex-shrink-0"
                  src="touch.svg"
                />
                <p>
                  controlAvatarTouch
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="hint.svg"
                />
                <p>
                  controlAvatarHint
                </p>
              </div>
            </section>
            <section
              class="space-y-3 mobile-landscape:space-y-2 portrait:space-y-2"
            >
              <h3
                class="font-bold text-adlerdarkblue"
              >
                openContent
              </h3>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="left-click.svg"
                />
                <p>
                  openContentMouse
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="left-click.svg"
                />
                <p>
                  openContentMouseDouble
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="touch.svg"
                />
                <p>
                  openContentTouch
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="hint.svg"
                />
                <div
                  class="flex flex-col gap-1"
                >
                  <p>
                    openContentHint1
                  </p>
                  <p>
                    openContentHint2
                  </p>
                </div>
              </div>
            </section>
            <section
              class="space-y-3 mobile-landscape:space-y-2 portrait:space-y-2"
            >
              <h3
                class="font-bold text-adlerdarkblue"
              >
                rotateCamera
              </h3>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="middle-mouse-click-and-drag.svg"
                />
                <p>
                  rotateCameraMouse
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="touch-and-drag.svg"
                />
                <p>
                  rotateCameraTouch
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="hint.svg"
                />
                <p>
                  rotateCameraHint
                </p>
              </div>
            </section>
            <section
              class="space-y-3 mobile-landscape:space-y-2 portrait:space-y-2"
            >
              <h3
                class="font-bold text-adlerdarkblue"
              >
                zoomCamera
              </h3>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="middle-mouse-click.svg"
                />
                <p>
                  zoomCameraMouse
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="touch-zoom.svg"
                />
                <p>
                  zoomCameraTouch
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="hint.svg"
                />
                <p>
                  zoomCameraHint
                </p>
              </div>
            </section>
            <section
              class="space-y-3 mobile-landscape:space-y-2 portrait:space-y-2"
            >
              <h3
                class="font-bold text-adlerdarkblue"
              >
                Highlight Farben
              </h3>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="LearningElementUnsolvedIcon.svg"
                />
                <p>
                  LearningElementUnsolvedIcon
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="LearningElementSolvedIcon.svg"
                />
                <p>
                  LearningElementSolvedIcon
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="NonLearningElementBaseIcon.svg"
                />
                <p>
                  NonLearningElementBaseIcon
                </p>
              </div>
              <div
                class="flex flex-row items-start justify-start gap-2"
              >
                <img
                  alt="Icon"
                  class="w-6 flex-shrink-0"
                  src="hint.svg"
                />
                <p>
                  HighlightHint
                </p>
              </div>
            </section>
          </div>
        </div>
      </div>
      <div
        class="flex-shrink-0 border-t border-adlerdarkblue/20 pt-4"
      >
        <p
          class="w-full text-center text-xs font-bold text-adlerdarkblue mobile-landscape:text-2xs portrait:text-2xs"
        >
          hint_controlsExplanationModal
        </p>
      </div>
    </div>
  </div>
</div>
`;
